### PR TITLE
[libxlsxwriter] Updated version 1.1.8

### PR DESCRIPTION
**Pull request by Shadowrom2020**
If this PR updates an existing port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] When updating the upstream version, the `"port-version"` is reset (removed from `vcpkg.json`).
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libxlsxwriter/dependencies.diff [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-index a87d080..fa58047 100644
---- a/dev/release/pkg-config.txt
-+++ b/dev/release/pkg-config.txt
-@@ -7,4 +7,5 @@ Name: libxlsxwriter
- Description: A C library for creating Excel XLSX files
- Version: @VERSION@
- Cflags: -I${includedir}
--Libs: -L${libdir} -lxlsxwriter -lz
-+Libs: -L${libdir} -lxlsxwriter
-+Requires.private: minizip zlib
```
Comment (dg0yt) [inline ports/libxlsxwriter/dependencies.diff:26]: We want to keep this.

**Review by dg0yt — COMMENTED** [review]
I think the port must now explicitly control interaction with PKGCONFIG when the build looks for dependencies. Disable it, or provide the program.

File: ports/libxlsxwriter/vcpkg.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "libxlsxwriter",
-  "version": "1.1.5",
-  "port-version": 2,
+  "version": "1.1.8",
+  "port-version": 3,
```
Comment (LilyWangLL) [inline ports/libxlsxwriter/vcpkg.json:4]:
```suggestion
```
Update port need to remove `port-version`.

**Review by LilyWangLL — DISMISSED** [review]

**Comment by dg0yt** [comment]
> I think the port must now explicitly control interaction with PKGCONFIG when the build looks for dependencies. Disable it, or provide the program

Still valid.

**Comment by Shadowrom2020** [comment]
> > I think the port must now explicitly control interaction with PKGCONFIG when the build looks for dependencies. Disable it, or provide the program
> 
> Still valid.

the port generated a pkg-config file but didn't use PKGCONFIG to find deps. I've removed the generation of the pkg-config file. The only reference to PKGCONFIG is in the Makefile, that isn't used to build the library when using CMake as build system.

**Comment by dg0yt** [comment]
This is not what was asked for. 
We had working pc files and CMake config, and there is no reson to damage that.

**Comment by Shadowrom2020** [comment]
> This is not what was asked for. We had working pc files and CMake config, and there is no reson to damage that.

Hmm... It works just fine now. When I updated the version to 1.1.8 it didn't build, because the old patches didn't work.
Could you please explain to me what you ment with:

> I think the port must now explicitly control interaction with PKGCONFIG when the build looks for dependencies. Disable it, or provide the program.

PKGCONFIG seems to be a optional feature introduced in 1.1.8. It's not used in building the library at all. I don't think it's necessary to provide pkg-config with the lib. I'm confused

**Comment by dg0yt** [comment]
- `CMakeLists.txt` has `find_package(PkgConfig)`. If it results in `PKG_CONFIG_FOUND`, it will configure using pkg-config, otherwise using CMake. 
- This is bad: On systems without pre-installed pkg-config, it may change the configuration result when port `pkgconf` is installed before `libxlsxwriter`. 
- So the task is to force exactly one behavior which works for all systems.
  - The pkg-config approach may work (with limited  won't support for downstream multiconfig usage). It requires that portfile uses `vcpkg_find_acquire_program(PKGCONFIG)` and passes `"-DPKG_CONFIG_EXECUTABLE=${PKGCONFIG}"` via cmake OPTIONS.
  - CMake may be enforced with `-DCMAKE_DISABLE_FIND_PACKAGE_PkgConfig=1`. It probably requires an adaptation of the original patching, for using unofficial packages and targets which carry complete transitive usage requirements.

IMO CMake is preferred when building with, and export config for, CMake.

**Comment by Shadowrom2020** [comment]
Ah, now I get the confusion.
I recon you looked at the main branch f the libxlsxwriter repository. Then you're correct, they seem to be working on PkgCongi integration for the next release.
But if you look at the v1.1.8 tag, you'll find there is no find_package(PkgConfig) in the CMakeLists.txt:

[CMakeLists.txt from v1.1.8](https://github.com/jmcnamara/libxlsxwriter/blob/v1.1.8/CMakeLists.txt)

**Comment by dg0yt** [comment]
Indeed. Usually updates are very close to master.
So updating our CMake patches might be even more relevant. Using unofficial exported CMake config from dependencies might be more maintainable for us. minizip may come with a bzip2 link dependency.

**Comment by Shadowrom2020** [comment]
I see... I've readded the unofficial::minizip patch

File: ports/libxlsxwriter/vcpkg.json (hunk 2 above)
Comment (Shadowrom2020) [reply ports/libxlsxwriter/vcpkg.json]: I removed the port-version, please re-review

File: ports/libxlsxwriter/dependencies.diff [hunk 3]
```diff
@@ -1,20 +1,5 @@
...
- # MINIZIP
- if (USE_SYSTEM_MINIZIP)
--    find_package(MINIZIP REQUIRED "1.0")
--    list(APPEND LXW_PRIVATE_INCLUDE_DIRS ${MINIZIP_INCLUDE_DIRS})
-+    find_package(MINIZIP NAMES unofficial-minizip REQUIRED)
-+    set(MINIZIP_LIBRARIES unofficial::minizip::minizip)
- endif()
- 
- # LIBRARY
 diff --git a/dev/release/pkg-config.txt b/dev/release/pkg-config.txt
```
Comment (dg0yt) [inline ports/libxlsxwriter/dependencies.diff:1]:
```suggestion
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0e855c77..b08ae8e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -230,8 +230,8 @@ message("zlib version: " ${ZLIB_VERSION})

 # MINIZIP
 if (USE_SYSTEM_MINIZIP)
-    find_package(MINIZIP "1.0" REQUIRED)
-    list(APPEND LXW_PRIVATE_INCLUDE_DIRS ${MINIZIP_INCLUDE_DIRS})
+    find_package(MINIZIP NAMES unofficial-minizip REQUIRED)
+    set(MINIZIP_LIBRARIES unofficial::minizip::minizip)
 endif()

 # LIBRARY
diff --git a/dev/release/pkg-config.txt b/dev/release/pkg-config.txt
```

I really want this (updated) chunk back in dependencies.diff. And nothing else from pkg-config.diff.

File: ports/libxlsxwriter/portfile.cmake [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-    REF "RELEASE_${VERSION}"
-    SHA512 bd7db0fcf25ebf492b4d8f7da8fdb6cc79400d7d0fa5856ddae259cb24817034fc97d4828cbde42434f41198dcfb6732ac63c756abd962689f4249ca64bf19c6
-    HEAD_REF master
+    REF "v${VERSION}"
+    SHA512 4d1df3b66e694629025ba4154a746d896f9fa32c727267cfbeacf72a3fc70d1b34c7bc767a03bca81395bbe2ff366fc4f4184c2c40126bc6b2d58b33a758cc8f
+    HEAD_REF main
     PATCHES
         dependencies.diff
-        fix-zlib-ver-required.patch
+        disable-pkgconfig.diff
```
Comment (dg0yt) [inline ports/libxlsxwriter/portfile.cmake:9]:
```suggestion
```

**Comment by dg0yt** [comment]
So https://github.com/microsoft/vcpkg/pull/41217#discussion_r1779199122 was left unresolved. 
 -> #41317

**Pull request merged**